R/tte_source.R

```diff
@@ -0,0 +1,256 @@
+#' Pre-Defined Time-to-Event Source Objects
+#'
+#' These pre-defined `tte_source` objects can be used as input to `derive_param_tte()`.
+#'
+#' @details
+#' To see the definition of the various objects simply print the object in the
+#' R console, e.g. `print(death_event)`.
+#'
+#' @seealso [derive_param_tte], [tte_source], [event_source], [censor_source]
+#'
+#' @export
+#'
+#' @keywords tte_source
+#'
+#' @rdname tte_source_objects
+#'
+death_event <- event_source(
+  dataset_name = "adrs",
+  filter = PARAMCD == "DEATH" & AVALC == "Y" & ANL01FL == "Y",
+  date = ADT,
+  set_values_to = vars(
+    EVNTDESC = "DEATH",
+    CNSDTDSC = "",
+    SRCDOM = "ADRS",
+    SRCVAR = "ADT",
+    SRCSEQ = ASEQ
+  )
+)
+
+#' @keywords tte_source
+#' @rdname tte_source_objects
+#' @export
+lastalive_censor <- censor_source(
+  dataset_name = "adsl",
+  date = LSTALVDT,
+  set_values_to = vars(
+    EVNTDESC = "ALIVE",
+    CNSDTDSC = "Alive During Study",
+    SRCDOM = "ADSL",
+    SRCVAR = "LSTALVDT",
+    SRCSEQ =
```

Comment:
@bundfussr also NA_character_ for numerics?

Reply:
I see there is `NA_integer_`, would we want `NA_integer_` or `NA_character_`?

Reply:
will defer to @bundfussr to comment when back Monday

Reply:
I would remove `SRCSEQ` from the definition instead of setting it to `NA_integer_`. Then the `SRCSEQ` variable is created only if at least one of the `tte_source` objects defines it. For example, if the TTE parameter is based on ADSL completely, the variable is not created. I think this makes more sense than setting it to `NA_integer_` for all observations.

Reply:
makes sense - thanks Stefan! @mattmarino - please just remove the line